lib/ddlog_std.rs

```diff
@@ -1514,7 +1514,7 @@ impl<V: Clone> Iterator for GroupIntoIter<V> {
         match self {
             GroupIntoIter::ByRef { iter, project } => match iter.next() {
                 None => None,
-                Some((x, w)) => Some(tuple2(project(x), *w as DDWeight)),
+                Some((x, w)) => Some(tuple2(project(x).clone(), *w as DDWeight)),
```

Comment:
I don't like all this cloning, do we have the ability to return references in ddlog right now?

Reply:
Agreed.  The clones were always there, hiding inside `project`. And yes, some of them can be removed now by changing various functions to return by-reference.  One unpleasant technicality is that we don't currently have a way to tell DDlog that group iterator returns `(&V, Weight)` tuples. We can do `(V, Weight)` or `&(V, Weight)`, but not `(&V, Weight)` or `(&V, &Weight)`.

I'll work on this once we've settled issues with the new semantics.

Reply:
Ok, this is trickier than I thought. Since this is not a regression, I'll merge as is and add a TODO in the code.